src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3401,11 +3405,19 @@ s32 Player_UpdateUpperBody(Player* this, PlayState* play) {
     return 1;
 }
 
-s32 func_80836898(PlayState* play, Player* this, PlayerFuncA74 func) {
-    this->func_A74 = func;
-    Player_SetupAction(play, this, Player_Action_808458D0, 0);
+/**
+ * This system exists to allow Player to delay a certain function until after an item in hand
+ * gets put away. The function provided by `putAwayDelayFunc` will only run after the put away
```

Comment:
I would call `putAwayDelayFunc` a callback?
maybe (but idk, didn't think much about it) `putAwayCallbackFunc`

Reply:
it is a callback yeah, but im not sure it needs to be in the name. ActionFucs are callbacks too for example and we dont feel the need to call them action callbacks

Reply:
without any other context I would see, in the context of oot
- "delay func" as an update function that waits for some condition before doing something (like an `EnThing_Wait`)
- "callback func" as a function that is called once some condition is met (somewhat rare in OoT afaict)

Reply:
as far as Im aware something is a callback when function pointers get passed and a function can "call back" on something else, like via a struct member or argument. Both this thing and action functions are callbacks by how theyre implemented

Reply:
yeah I'm just saying to me "delay func" doesn't sound like a one-time (discrete event) callback as much as "callback func" would
"one-time callback" as opposed to "every-frame callback" aka action/update/draw funcs

Reply:
That is fair. I see the point, but dont think callback is the word we are looking for to make that distinction. 

Technically speaking this would run every frame, but the intent here is that the "delay func" is a function that sets up a new action and moves execution away from `Player_Action_PutAwayDelay`. As far as I know every use case of this does only run once, but thats only because of what the delay funcs themselves do, not the system itself

Reply:
Due to conversations with cadmic below, the names are much different now. Is this point still a problem for you with these new names?